generators/app/templates/gulp/tasks/report.js

```diff
@@ -0,0 +1,233 @@
+
+function listFiles({ destBase }) {
+  return through.obj(function getInfoAboutFile(file, enc, callback) {
+    const relativePath = path.relative(destBase, file.path);
+    const bufferContents = Buffer.isBuffer(file.contents)
+      ? file.contents
+      : Buffer.from(file.contents);
+
+    files[relativePath] = {
+      path: file.path,
+      hasSourceMap: false,
+      size: bufferContents.length,
+      sizeGzipped: -1,
+      notes: [],
+    };
+
+    zlib.gzip(bufferContents, (err, zipped) => {
+      if (err) {
+        throw err;
+      }
+
+      files[relativePath].sizeGzipped = zipped.length;
+      callback();
+    });
+
+    this.push(file);
+  });
+}
+
+function markSouceMaps({ destBase }) {
+  return through.obj((file, enc, callback) => {
+    const relativePath = path.relative(destBase, file.path);
+
+    if (!files[relativePath]) {
+      return;
+    }
+
+    if (file.sourceMap && file.sourceMap.preExistingComment) {
+      if (file.sourceMap.sources.length > 1) {
+        files[relativePath].hasSourceMap = true;
+      } else {
+        files[relativePath].notes.push('only one file');
+      }
+    }
+
+    callback();
+  });
+}
+
+function escapeHTML(s) {
+  return s
+    .replace(/&/g, '&amp;')
+    .replace(/"/g, '&quot;')
+    .replace(/</g, '&lt;')
+    .replace(/>/g, '&gt;');
+}
+
+function hasFile(resolvedPath) {
+  if (!resolvedPath.endsWith('.html')) {
+    return false;
+  }
+
+  const sourcePath = resolvedPath.slice(0, -5);
+
+  return fs.existsSync(sourcePath);
+}
+
+function formatSize(n) {
+  if (n < 1024) {
+    return `${n} B`;
+  } else if (n < MEBIBYTE) {
+    return `${(n / 1024).toFixed(2)} KiB`;
+  }
+  return `${(n / MEBIBYTE).toFixed(2)} MiB`;
+}
+
+// https://github.com/shakyShane/dev-ip/blob/9f5a1b6154a16db88ca276c08426867c55924e61/lib/dev-ip.js
+function getIp() {
+  const networkInterfaces = os.networkInterfaces();
+  const matches = [];
+
+  Object.keys(networkInterfaces).forEach(item => {
+    networkInterfaces[item].forEach(address => {
+      if (address.internal === false && address.family === 'IPv4') {
+        matches.push(address.address);
+      }
+    });
+  });
+
+  return matches;
+}
+
+function startServer({ destBase }) {
+  const root = path.resolve(destBase);
+  const server = http.createServer((req, res) => {
+    const { url } = req;
+    const resolvedPath = path.resolve(root, path.join(root, url));
+
+    if (!resolvedPath.startsWith(root)) {
+      res.writeHead(403);
+      res.end();
+      return;
+    }
+
+    if (url === '/') {
+      res.writeHead(200, {
+        'Content-Type': 'text/html; charset=utf-8',
+      });
+
+      res.write(
+        `<!doctype html><html lang="en"><title>Chisel SourceMap Reports</title>
+        <table><thead><tr><td>Name</td><td>Size</td><td>Gzipped size</td>
+        <td>Notes</td></tr></thead><tbody>`
+      );
+
+      const links = [];
+
+      Object.keys(files)
+        .sort()
+        .forEach(link => {
```

Comment:
Hmm. Maybe `.map` instead of pushing using `forEach`? You could return a string with `tr`. A bit less writing. Unless I missed something. 🤔

Reply:
Yep, done.